framework/src/utils/MeshBaseDiagnosticsUtils.C

```diff
@@ -66,4 +67,93 @@ checkNonConformalMesh(const std::unique_ptr<MeshBase> & mesh,
   }
   pl->unset_close_to_point_tol();
 }
+
+bool
+checkFirstOrderEdgeOverlap(const std::unique_ptr<Elem> & edge1,
+                           const std::unique_ptr<Elem> & edge2,
+                           std::vector<double> & intersection_coords,
+                           const Real intersection_tol)
+{
+  // check that the two elements are of type EDGE2
+  mooseAssert(edge1->type() == EDGE2, "Elements must be of type EDGE2");
+  mooseAssert(edge2->type() == EDGE2, "Elements must be of type EDGE2");
+  // Get nodes from the two edges
+  const Node * const n1 = edge1->get_nodes()[0];
+  const Node * const n2 = edge1->get_nodes()[1];
+  const Node * const n3 = edge2->get_nodes()[0];
+  const Node * const n4 = edge2->get_nodes()[1];
+
+  // get x,y,z coordinates for each node
+  const auto n1x = n1->operator()(0);
+  const auto n1y = n1->operator()(1);
+  const auto n1z = n1->operator()(2);
+  const auto n2x = n2->operator()(0);
+  const auto n2y = n2->operator()(1);
+  const auto n2z = n2->operator()(2);
+  const auto n3x = n3->operator()(0);
+  const auto n3y = n3->operator()(1);
+  const auto n3z = n3->operator()(2);
+  const auto n4x = n4->operator()(0);
+  const auto n4y = n4->operator()(1);
+  const auto n4z = n4->operator()(2);
+
+  // Check that the two edges are not sharing a node
+  if (n1->id() == n3->id() || n1->id() == n4->id() || n2->id() == n3->id() || n2->id() == n4->id())
+    return false;
+
+  /*
+    There's a chance that they overlap. Find shortest line that connects two edges and if its length
+    is close enough to 0 return true The shortest line between the two edges will be perpendicular
+    to both.
+  */
+  const auto d1343 =
+      (n1x - n3x) * (n4x - n3x) + (n1y - n3y) * (n4y - n3y) + (n1z - n3z) * (n4z - n3z);
+  const auto d4321 =
+      (n4x - n3x) * (n2x - n1x) + (n4y - n3y) * (n2y - n1y) + (n4z - n3z) * (n2z - n1z);
+  const auto d1321 =
+      (n1x - n3x) * (n2x - n1x) + (n1y - n3y) * (n2y - n1y) + (n1z - n3z) * (n2z - n1z);
+  const auto d4343 =
```

Comment:
For that one it works out that way. I was thinking it may be better to keep it as is so that it's more clear that it's doing the same thing as the other 3 calculations?

Reply:
Or I could rewrite it as a dot product like the rest?

Reply:
I'd still prefer the `norm_sq()` but a dot product would be fine for the symmetry.